zap/src/irgen/des.rs

```diff
@@ -238,7 +238,11 @@ impl Des<'_> {
 						self.push_range_check(len_expr.clone(), *range);
 					}
 
-					self.push_assign(into, self.readstring(len_expr.clone()));
+					self.push_assign(into.clone(), self.readstring(len_expr.clone()));
+
+					if *utf8 {
+						self.push_utf8_check(Expr::Var(into.into()));
+					}
```

Comment:
I think this is the only place (unlike array) where we assign to `into` before validating... I think that is okay though?